connect-inject/handler.go

```diff
@@ -393,6 +478,76 @@ func (h *Handler) defaultAnnotations(pod *corev1.Pod, patches *[]jsonpatch.JsonP
 	return nil
 }
 
+func (h *Handler) consulNamespace(ns string) string {
+	// Return an empty string if namespaces aren't enabled. This
+	// functionality is used for init and sidecar container creation.
+	if !h.EnableNamespaces {
+		return ""
+	}
+
+	// Mirroring takes precedence
+	if h.EnableNSMirroring {
+		return fmt.Sprintf("%s%s", h.MirroringPrefix, ns)
+	} else {
+		return h.ConsulNSName
+	}
+}
+
+func (h *Handler) checkAndCreateNamespace(ns string) error {
+	// Check if the Consul namespace exists
+	namespaceInfo, _, err := h.ConsulClient.Namespaces().Read(ns, nil)
+	if err != nil {
+		return err
+	}
+
+	// If not, create it
+	if namespaceInfo == nil {
+		consulNamespace := api.Namespace{
+			Name:        ns,
+			Description: "Auto-generated by a Connect Injector",
+			// TODO: when metadata is added to the api
+			// Meta:        map[string]string{"external-source": "kubernetes"},
+		}
+
+		_, _, err = h.ConsulClient.Namespaces().Create(&consulNamespace, nil)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (h *Handler) checkAndCreateBindingRule(ns string) error {
+	// Check if the binding rule exists in this namespace
```

Comment:
Was a bit confused by this comment because my immediate thought was "what if there's more than one binding rule in this namespace". I realized after a bit that you're only checking for binding rules for our auth method so maybe you can clarify that.

Reply:
Updated.